THCBlas error building from source

Hi, I was trying to update from source but I get an error "no suitable constructor exists to convert from "int" to "__half"" when trying to build pytorch/aten/src/THC/THCBlas.cu.

I am working with python 2.7 cuda 7.5 and GCC 4.8.3 in openSUSE. Any suggest?

Here is the output error when building from source:
---------------------------------------------------------------------------------------------------------------------------------
ATen Excluded: set(['bernoulli', 'bernoulli_'])
[  0%] Built target aten_files_are_generated
[  0%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCReduceApplyUtils.cu.o
[  1%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCStorage.cu.o
[  1%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCSleep.cu.o
[  1%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCBlas.cu.o
[  1%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCStorageCopy.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensor.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorMath.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorCopy.cu.o
/home/gabriel/pytorch/aten/src/THC/THCBlas.cu(77): error: no suitable constructor exists to convert from "int" to "__half"

/home/gabriel/pytorch/aten/src/THC/THCBlas.cu(77): error: no suitable constructor exists to convert from "float" to "__half"

2 errors detected in the compilation of "/tmp/tmpxft_0000587a_00000000-7_THCBlas.cpp1.ii".
CMake Error at ATen_generated_THCBlas.cu.o.cmake:267 (message):
  Error generating file
  /home/gabriel/pytorch/torch/lib/build/aten/src/ATen/CMakeFiles/ATen.dir/__/THC/./ATen_generated_THCBlas.cu.o


src/ATen/CMakeFiles/ATen.dir/build.make:70: recipe for target 'src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCBlas.cu.o' failed
make[2]: *** [src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCBlas.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:194: recipe for target 'src/ATen/CMakeFiles/ATen.dir/all' failed
make[1]: *** [src/ATen/CMakeFiles/ATen.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

Same problem here. 

CUDA: v7.5
CUDNN: v6.0

```
running install
running build_deps
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /home/zhangruiqing01/tools/anaconda3/bin/cc
-- Check for working C compiler: /home/zhangruiqing01/tools/anaconda3/bin/cc -- works
-- Detecting C compiler ABI info
-- Checking if C linker supports --verbose
-- Checking if C linker supports --verbose - yes
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/zhangruiqing01/tools/anaconda3/bin/c++
-- Check for working CXX compiler: /home/zhangruiqing01/tools/anaconda3/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Checking if CXX linker supports --verbose
-- Checking if CXX linker supports --verbose - yes
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found CUDA: /usr/local/cuda (found suitable version "7.5", minimum required is "7.0") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl
/usr/bin/make64 MAC=64 install
Scanning dependencies of target nccl
[100%] Generating lib/libnccl.so
/usr/bin/make64 MAC=64 -j 12
Grabbing  src/nccl.h                          > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/include/nccl.h
Compiling src/libwrap.cu                      > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/obj/libwrap.o
Compiling src/core.cu                         > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/obj/core.o
Compiling src/all_gather.cu                   > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/obj/all_gather.o
Compiling src/all_reduce.cu                   > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/obj/all_reduce.o
Compiling src/broadcast.cu                    > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/obj/broadcast.o
Compiling src/reduce.cu                       > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/obj/reduce.o
Compiling src/reduce_scatter.cu               > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/obj/reduce_scatter.o
ptxas warning : Too big maxrregcount value specified 96, will be ignored
ptxas warning : Too big maxrregcount value specified 96, will be ignored
ptxas warning : Too big maxrregcount value specified 96, will be ignored
ptxas warning : Too big maxrregcount value specified 96, will be ignored
ptxas warning : Too big maxrregcount value specified 96, will be ignored
ptxas warning : Too big maxrregcount value specified 96, will be ignored
ptxas warning : Too big maxrregcount value specified 96, will be ignored
Linking   libnccl.so.1.3.5                    > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/lib/libnccl.so.1.3.5
Archiving libnccl_static.a                    > /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/nccl/lib/libnccl_static.a
[100%] Built target nccl
Install the project...
-- Install configuration: "Release"
-- Installing: /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/tmp_install/include/nccl.h
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /home/zhangruiqing01/tools/anaconda3/bin/cc
-- Check for working C compiler: /home/zhangruiqing01/tools/anaconda3/bin/cc -- works
-- Detecting C compiler ABI info
-- Checking if C linker supports --verbose
-- Checking if C linker supports --verbose - yes
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/zhangruiqing01/tools/anaconda3/bin/c++
-- Check for working CXX compiler: /home/zhangruiqing01/tools/anaconda3/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Checking if CXX linker supports --verbose
-- Checking if CXX linker supports --verbose - yes
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found CUDA: /usr/local/cuda (found suitable version "7.5", minimum required is "5.5") 
-- Autodetected CUDA architecture(s): 3.5 3.5 3.5 3.5
-- Found CUDA with FP16 support, compiling with torch.CudaHalfTensor
-- Removing -DNDEBUG from compile flags
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp  
-- Compiling with OpenMP support
-- Checking prototype magma_get_sgeqrf_nb for MAGMA_V2 - True
-- Compiling with MAGMA support
-- MAGMA INCLUDE DIRECTORIES: /home/zhangruiqing01/tools/anaconda3/include
-- MAGMA LIBRARIES: /home/zhangruiqing01/tools/anaconda3/lib/libmagma.a
-- MAGMA V2 check: 1
-- Could not find hardware support for NEON on this machine.
-- No OMAP3 processor on this machine.
-- No OMAP4 processor on this machine.
-- Looking for cpuid.h
-- Looking for cpuid.h - found
-- Performing Test HAVE_GCC_GET_CPUID
-- Performing Test HAVE_GCC_GET_CPUID - Success
-- Performing Test NO_GCC_EBX_FPIC_BUG
-- Performing Test NO_GCC_EBX_FPIC_BUG - Success
-- Performing Test C_HAS_SSE1_1
-- Performing Test C_HAS_SSE1_1 - Success
-- Performing Test C_HAS_SSE2_1
-- Performing Test C_HAS_SSE2_1 - Success
-- Performing Test C_HAS_SSE3_1
-- Performing Test C_HAS_SSE3_1 - Failed
-- Performing Test C_HAS_SSE3_2
-- Performing Test C_HAS_SSE3_2 - Success
-- Performing Test C_HAS_SSE4_1_1
-- Performing Test C_HAS_SSE4_1_1 - Failed
-- Performing Test C_HAS_SSE4_1_2
-- Performing Test C_HAS_SSE4_1_2 - Success
-- Performing Test C_HAS_SSE4_2_1
-- Performing Test C_HAS_SSE4_2_1 - Failed
-- Performing Test C_HAS_SSE4_2_2
-- Performing Test C_HAS_SSE4_2_2 - Success
-- Performing Test C_HAS_AVX_1
-- Performing Test C_HAS_AVX_1 - Failed
-- Performing Test C_HAS_AVX_2
-- Performing Test C_HAS_AVX_2 - Success
-- Performing Test C_HAS_AVX2_1
-- Performing Test C_HAS_AVX2_1 - Failed
-- Performing Test C_HAS_AVX2_2
-- Performing Test C_HAS_AVX2_2 - Failed
-- Performing Test C_HAS_AVX2_3
-- Performing Test C_HAS_AVX2_3 - Failed
-- Performing Test CXX_HAS_SSE1_1
-- Performing Test CXX_HAS_SSE1_1 - Success
-- Performing Test CXX_HAS_SSE2_1
-- Performing Test CXX_HAS_SSE2_1 - Success
-- Performing Test CXX_HAS_SSE3_2 - Success
-- Performing Test CXX_HAS_SSE4_1_1
-- Performing Test CXX_HAS_SSE4_1_1 - Failed
-- Performing Test CXX_HAS_SSE4_1_2
-- Performing Test CXX_HAS_SSE4_1_2 - Success
-- Performing Test CXX_HAS_SSE4_2_1
-- Performing Test CXX_HAS_SSE4_2_1 - Failed
-- Performing Test CXX_HAS_SSE4_2_2
-- Performing Test CXX_HAS_SSE4_2_2 - Success
-- Performing Test CXX_HAS_AVX_1
-- Performing Test CXX_HAS_AVX_1 - Failed
-- Performing Test CXX_HAS_AVX_2
-- Performing Test CXX_HAS_AVX_2 - Success
-- Performing Test CXX_HAS_AVX2_1
-- Performing Test CXX_HAS_AVX2_1 - Failed
-- Performing Test CXX_HAS_AVX2_2
-- Performing Test CXX_HAS_AVX2_2 - Failed
-- Performing Test CXX_HAS_AVX2_3
-- Performing Test CXX_HAS_AVX2_3 - Failed
-- SSE2 Found
-- SSE3 Found
-- AVX Found
-- Performing Test HAS_C11_ATOMICS
-- Performing Test HAS_C11_ATOMICS - Failed
-- Performing Test HAS_MSC_ATOMICS
-- Performing Test HAS_MSC_ATOMICS - Failed
-- Performing Test HAS_GCC_ATOMICS
-- Performing Test HAS_GCC_ATOMICS - Success
-- Atomics: using GCC intrinsics
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of void*
-- Check size of void* - done
-- Checking for [mkl_gf_lp64 - mkl_gnu_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_gf_lp64: not found
-- Checking for [mkl_gf_lp64 - mkl_intel_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_gf_lp64: not found
-- Checking for [mkl_gf - mkl_gnu_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_gf: not found
-- Checking for [mkl_gf - mkl_intel_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_gf: not found
-- Checking for [mkl_intel_lp64 - mkl_gnu_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_intel_lp64: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_intel_lp64.so
--   Library mkl_gnu_thread: not found
-- Checking for [mkl_intel_lp64 - mkl_intel_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_intel_lp64: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_intel_lp64.so
--   Library mkl_intel_thread: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_intel_thread.so
--   Library mkl_core: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_core.so
--   Library gomp: -fopenmp
--   Library pthread: /usr/lib64/libpthread.so
--   Library m: /usr/lib64/libm.so
--   Library dl: /usr/lib64/libdl.so
-- Looking for cblas_sgemm
-- Looking for cblas_sgemm - not found
-- Checking for [mkl_intel - mkl_gnu_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_intel: not found
-- Checking for [mkl_intel - mkl_intel_thread - mkl_core - gomp - pthread - m - dl]
--   Library mkl_intel: not found
-- Checking for [mkl_gf_lp64 - mkl_gnu_thread - mkl_core - iomp5 - pthread - m - dl]
--   Library mkl_gf_lp64: not found
-- Checking for [mkl_gf_lp64 - mkl_intel_thread - mkl_core - iomp5 - pthread - m - dl]
--   Library mkl_gf_lp64: not found
-- Checking for [mkl_gf - mkl_gnu_thread - mkl_core - iomp5 - pthread - m - dl]
--   Library mkl_gf: not found
-- Checking for [mkl_gf - mkl_intel_thread - mkl_core - iomp5 - pthread - m - dl]
--   Library mkl_gf: not found
-- Checking for [mkl_intel_lp64 - mkl_gnu_thread - mkl_core - iomp5 - pthread - m - dl]
--   Library mkl_intel_lp64: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_intel_lp64.so
--   Library mkl_gnu_thread: not found
-- Checking for [mkl_intel_lp64 - mkl_intel_thread - mkl_core - iomp5 - pthread - m - dl]
--   Library mkl_intel_lp64: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_intel_lp64.so
--   Library mkl_intel_thread: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_intel_thread.so
--   Library mkl_core: /home/zhangruiqing01/tools/anaconda3/lib/libmkl_core.so
--   Library iomp5: /home/zhangruiqing01/tools/anaconda3/lib/libiomp5.so
--   Library pthread: /usr/lib64/libpthread.so
--   Library m: /usr/lib64/libm.so
--   Library dl: /usr/lib64/libdl.so
-- Looking for cblas_sgemm
-- Looking for cblas_sgemm - found
-- MKL library found
-- Performing Test BLAS_F2C_DOUBLE_WORKS
-- Performing Test BLAS_F2C_DOUBLE_WORKS - Failed
-- Performing Test BLAS_F2C_FLOAT_WORKS
-- Performing Test BLAS_F2C_FLOAT_WORKS - Success
-- Performing Test BLAS_USE_CBLAS_DOT
-- Performing Test BLAS_USE_CBLAS_DOT - Success
-- Found a library with BLAS API (mkl).
-- Found a library with LAPACK API. (mkl)
-- Found CUDNN: /home/zhangruiqing01/tools/py-torch-install/cudnn6.0/include  
-- Found cuDNN: v6.0.21  (include: /home/zhangruiqing01/tools/py-torch-install/cudnn6.0/include, library: /home/zhangruiqing01/tools/py-torch-install/cudnn6.0/lib64/libcudnn.so)
-- Could NOT find NNPACK (missing:  NNPACK_INCLUDE_DIR NNPACK_LIBRARY CPUINFO_LIBRARY PTHREADPOOL_LIBRARY) 
-- NNPACK not found. Compiling without nNPACK support
-- Using python found in /home/zhangruiqing01/tools/anaconda3/bin/python
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Looking for mmap
-- Looking for mmap - found
-- Looking for shm_open
-- Looking for shm_open - found
-- Looking for shm_unlink
-- Looking for shm_unlink - found
-- Looking for malloc_usable_size
-- Looking for malloc_usable_size - found
-- Performing Test C_HAS_THREAD
-- Performing Test C_HAS_THREAD - Success
disable contrib because ATEN_NO_CONTRIB is set
-- Configuring done
-- Generating done
-- Build files have been written to: /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/aten
/usr/bin/make64 MAC=64 install -j12
Scanning dependencies of target aten_files_are_generated
[  0%] Generating ATen/CPUByteStorage.cpp, ATen/CPUByteStorage.h, ATen/CPUByteTensor.cpp, ATen/CPUByteTensor.h, ATen/CPUByteType.cpp, ATen/CPUByteType.h, ATen/CPUCharStorage.cpp, ATen/CPUCharStorage.h, ATen/CPUCharTensor.cpp, ATen/CPUCharTensor.h, ATen/CPUCharType.cpp, ATen/CPUCharType.h, ATen/CPUDoubleStorage.cpp, ATen/CPUDoubleStorage.h, ATen/CPUDoubleTensor.cpp, ATen/CPUDoubleTensor.h, ATen/CPUDoubleType.cpp, ATen/CPUDoubleType.h, ATen/CPUFloatStorage.cpp, ATen/CPUFloatStorage.h, ATen/CPUFloatTensor.cpp, ATen/CPUFloatTensor.h, ATen/CPUFloatType.cpp, ATen/CPUFloatType.h, ATen/CPUGenerator.h, ATen/CPUHalfStorage.cpp, ATen/CPUHalfStorage.h, ATen/CPUHalfTensor.cpp, ATen/CPUHalfTensor.h, ATen/CPUHalfType.cpp, ATen/CPUHalfType.h, ATen/CPUIntStorage.cpp, ATen/CPUIntStorage.h, ATen/CPUIntTensor.cpp, ATen/CPUIntTensor.h, ATen/CPUIntType.cpp, ATen/CPUIntType.h, ATen/CPULongStorage.cpp, ATen/CPULongStorage.h, ATen/CPULongTensor.cpp, ATen/CPULongTensor.h, ATen/CPULongType.cpp, ATen/CPULongType.h, ATen/CPUShortStorage.cpp, ATen/CPUShortStorage.h, ATen/CPUShortTensor.cpp, ATen/CPUShortTensor.h, ATen/CPUShortType.cpp, ATen/CPUShortType.h, ATen/CUDAByteStorage.cpp, ATen/CUDAByteStorage.h, ATen/CUDAByteTensor.cpp, ATen/CUDAByteTensor.h, ATen/CUDAByteType.cpp, ATen/CUDAByteType.h, ATen/CUDACharStorage.cpp, ATen/CUDACharStorage.h, ATen/CUDACharTensor.cpp, ATen/CUDACharTensor.h, ATen/CUDACharType.cpp, ATen/CUDACharType.h, ATen/CUDADoubleStorage.cpp, ATen/CUDADoubleStorage.h, ATen/CUDADoubleTensor.cpp, ATen/CUDADoubleTensor.h, ATen/CUDADoubleType.cpp, ATen/CUDADoubleType.h, ATen/CUDAFloatStorage.cpp, ATen/CUDAFloatStorage.h, ATen/CUDAFloatTensor.cpp, ATen/CUDAFloatTensor.h, ATen/CUDAFloatType.cpp, ATen/CUDAFloatType.h, ATen/CUDAGenerator.h, ATen/CUDAHalfStorage.cpp, ATen/CUDAHalfStorage.h, ATen/CUDAHalfTensor.cpp, ATen/CUDAHalfTensor.h, ATen/CUDAHalfType.cpp, ATen/CUDAHalfType.h, ATen/CUDAIntStorage.cpp, ATen/CUDAIntStorage.h, ATen/CUDAIntTensor.cpp, ATen/CUDAIntTensor.h, ATen/CUDAIntType.cpp, ATen/CUDAIntType.h, ATen/CUDALongStorage.cpp, ATen/CUDALongStorage.h, ATen/CUDALongTensor.cpp, ATen/CUDALongTensor.h, ATen/CUDALongType.cpp, ATen/CUDALongType.h, ATen/CUDAShortStorage.cpp, ATen/CUDAShortStorage.h, ATen/CUDAShortTensor.cpp, ATen/CUDAShortTensor.h, ATen/CUDAShortType.cpp, ATen/CUDAShortType.h, ATen/Copy.cpp, ATen/Declarations.yaml, ATen/Functions.h, ATen/NativeFunctions.h, ATen/SparseCPUByteTensor.cpp, ATen/SparseCPUByteTensor.h, ATen/SparseCPUByteType.cpp, ATen/SparseCPUByteType.h, ATen/SparseCPUCharTensor.cpp, ATen/SparseCPUCharTensor.h, ATen/SparseCPUCharType.cpp, ATen/SparseCPUCharType.h, ATen/SparseCPUDoubleTensor.cpp, ATen/SparseCPUDoubleTensor.h, ATen/SparseCPUDoubleType.cpp, ATen/SparseCPUDoubleType.h, ATen/SparseCPUFloatTensor.cpp, ATen/SparseCPUFloatTensor.h, ATen/SparseCPUFloatType.cpp, ATen/SparseCPUFloatType.h, ATen/SparseCPUIntTensor.cpp, ATen/SparseCPUIntTensor.h, ATen/SparseCPUIntType.cpp, ATen/SparseCPUIntType.h, ATen/SparseCPULongTensor.cpp, ATen/SparseCPULongTensor.h, ATen/SparseCPULongType.cpp, ATen/SparseCPULongType.h, ATen/SparseCPUShortTensor.cpp, ATen/SparseCPUShortTensor.h, ATen/SparseCPUShortType.cpp, ATen/SparseCPUShortType.h, ATen/SparseCUDAByteTensor.cpp, ATen/SparseCUDAByteTensor.h, ATen/SparseCUDAByteType.cpp, ATen/SparseCUDAByteType.h, ATen/SparseCUDACharTensor.cpp, ATen/SparseCUDACharTensor.h, ATen/SparseCUDACharType.cpp, ATen/SparseCUDACharType.h, ATen/SparseCUDADoubleTensor.cpp, ATen/SparseCUDADoubleTensor.h, ATen/SparseCUDADoubleType.cpp, ATen/SparseCUDADoubleType.h, ATen/SparseCUDAFloatTensor.cpp, ATen/SparseCUDAFloatTensor.h, ATen/SparseCUDAFloatType.cpp, ATen/SparseCUDAFloatType.h, ATen/SparseCUDAIntTensor.cpp, ATen/SparseCUDAIntTensor.h, ATen/SparseCUDAIntType.cpp, ATen/SparseCUDAIntType.h, ATen/SparseCUDALongTensor.cpp, ATen/SparseCUDALongTensor.h, ATen/SparseCUDALongType.cpp, ATen/SparseCUDALongType.h, ATen/SparseCUDAShortTensor.cpp, ATen/SparseCUDAShortTensor.h, ATen/SparseCUDAShortType.cpp, ATen/SparseCUDAShortType.h, ATen/Tensor.h, ATen/TensorMethods.h, ATen/Type.cpp, ATen/Type.h
[  0%] Built target aten_files_are_generated
[  0%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCReduceApplyUtils.cu.o
[  0%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCBlas.cu.o
[  0%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCSleep.cu.o
[  1%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCStorage.cu.o
[  1%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCStorageCopy.cu.o
[  1%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensor.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorCopy.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorMath.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorMathBlas.cu.o
[  3%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorMathMagma.cu.o
[  3%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorMathPairwise.cu.o
[  3%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorMathReduce.cu.o
/home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/aten/src/THC/THCBlas.cu(77): error: no suitable constructor exists to convert from "int" to "__half"

/home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/aten/src/THC/THCBlas.cu(77): error: no suitable constructor exists to convert from "float" to "__half"

2 errors detected in the compilation of "/tmp/tmpxft_000027da_00000000-7_THCBlas.cpp1.ii".
CMake Error at ATen_generated_THCBlas.cu.o.cmake:267 (message):
  Error generating file
  /home/zhangruiqing01/tools/py-torch-install/pytorch/pytorch/torch/lib/build/aten/src/ATen/CMakeFiles/ATen.dir/__/THC/./ATen_generated_THCBlas.cu.o


make64[2]: *** [src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCBlas.cu.o] Error 1
make64[2]: *** Waiting for unfinished jobs....
make64[1]: *** [src/ATen/CMakeFiles/ATen.dir/all] Error 2
make64: *** [all] Error 2
[zhangruiqing01@yq01-idl-gpu-offline41.yq01.baidu.com:~/tools/py-torch-install/pytorch/pytorch]
$ 
```

--- a/aten/src/THC/THCBlas.cu
+++ b/aten/src/THC/THCBlas.cu
@@ -74,7 +74,7 @@ half THCudaBlas_Hdot(THCState *state, int64_t n, half *x, int64_t incx, half *y,
   return THC_float2half(0);
 #else
   THError("Cublas_Hdot requires CUDA 8.0+");
-  return THC_half2float(0);
+  return THC_float2half(0);
 #endif
 }
 #endif

